### PR TITLE
User p

### PR DESCRIPTION
**Pull request by rubelux**
This PR included editing and creating user, new data schema an update ```newProfile``` component and form

### CODE REVIEW

**Review by EricWVGG — APPROVED** [review]
I'm not crazy about the way we're created this `finalSchema` and `finalUiSchema` thing. I'd prefer that we instead did something like…

```
import { schema, uischema, adminSchema, adminUiSchema, FormState } from '@forms/createProfile'

...

<Form
  schema={modalAttrs.scope !== 'admin' ? schema : adminSchema}
  uischema={modalAttrs.scope !== 'admin' ? adminUiSchema : uischema}
  {...{ data, etc. }}
/>
```

Then we could ditch all that conditional logic in the component and make everything a lot cleaner.

You could even use the exact same logic to prep the admin schema, just move that logic to a adminSchema.ts file.

```
import schema from './schema`

export default {
  ...schema,
  ... {
    properties: {
  }
}
```

regardless, this looks fine, we can chalk that up to a to-do for later

File: src/components/CreateProfile/index.tsx [hunk 1]
```diff
@@ -5,24 +5,103 @@ import Link from 'next/link'
...
+              publicAddress: {
+                type: 'string',
+                minLength: 2,
+              },
+            },
+          },
+        }
+      : schema
+
+  const updateProfileC = async () => {
```
[the remark this answers is not in the document]
Comment (rubelux) [reply src/components/CreateProfile/index.tsx]: I updated the name, not sure about the second point

File: src/components/CreateProfile/lib/index.ts [hunk 2]
```diff
@@ -44,19 +46,64 @@ const useCreateProfile = () => {
...
+        ],
+      },
+    })
+    await addUserToCourier()
+
+    if (!createUserR.data?.insert_Users) {
+      throw new Error('Error creating users in the admin form')
+    }
+
+    return createUserR.data?.insert_Users?.returning[0]
```
[the remark this answers is not in the document]
Comment (rubelux) [reply src/components/CreateProfile/lib/index.ts:70]: removed them

File: src/components/Layout/Modals/NewProjectMembersModal/index.tsx [hunk 3]
```diff
@@ -45,18 +79,51 @@ const NewProjectMembersModal = () => {
...
-        <Button level={2} outline onClick={() => setuserSelection(null)}>
-          Create New User
-        </Button>
+        {!userSelected && (
+          <Button
+            level={2}
+            outline
+            onClick={() => {
+              toggleModal()
+              setVisibleModalWithAttrs('createProfile', {
```
[the remark this answers is not in the document]
Comment (rubelux) [reply src/components/Layout/Modals/NewProjectMembersModal/index.tsx]: Moved all the event handler to its own arrow fn

File: src/components/CreateProfile/index.tsx [hunk 4]
```diff
@@ -5,24 +5,103 @@ import Link from 'next/link'
...
+        externalLink: modalAttrs?.initialState?.externalLink,
+        wallet: modalAttrs?.initialState?.wallet,
+        publicAddress: modalAttrs?.initialState?.publicAddress,
+      }
+    : {}
+
+  const { updateProfile, createProfile, additionalErrors, data, setData, setImageFile } =
+    useCreateProfile(newInitialState)
+
+  const finalUischema =
```
[the remark this answers is not in the document]
Comment (rubelux) [reply src/components/CreateProfile/index.tsx]: As suggested in the previous comment, I removed all this and moved the `schema` and `uiSchema` to the form config file

**Pull request merged**